application_sdk/services/objectstore.py

```diff
@@ -279,13 +292,31 @@ async def upload_file(
             raise e
 
         try:
+            # Primary upload to the specified store
             await cls._invoke_dapr_binding(
                 operation=cls.OBJECT_CREATE_OPERATION,
                 data=file_content,
                 metadata=cls._create_file_metadata(destination),
                 store_name=store_name,
             )
-            logger.debug(f"Successfully uploaded file: {destination}")
+            logger.debug(f"Successfully uploaded file to {store_name}: {destination}")
+            
+            # Secondary upload to upstream store if enabled
+            if ENABLE_ATLAN_UPLOAD and store_name != UPSTREAM_OBJECT_STORE_NAME:
+                try:
+                    await cls._invoke_dapr_binding(
+                        operation=cls.OBJECT_CREATE_OPERATION,
+                        data=file_content,
+                        metadata=cls._create_file_metadata(destination),
+                        store_name=UPSTREAM_OBJECT_STORE_NAME,
+                    )
+                    logger.debug(f"Successfully uploaded file to upstream store {UPSTREAM_OBJECT_STORE_NAME}: {destination}")
+                except Exception as upstream_error:
+                    # Log upstream upload error but don't fail the entire operation
+                    logger.warning(
+                        f"Failed to upload file to upstream store {UPSTREAM_OBJECT_STORE_NAME}: {str(upstream_error)}"
+                    )
```

Comment:
we should not be doing this here - it should rather be in the output/parquet.py behind a condition